File: MyReivces/app/src/main/java/com/fire/myreivces/ui/designMod/java/Animal.java
package com.fire.myreivces.ui.designMod.java;

import android.util.Log;

public class Animal implements Fly, Run{

  private final String t = "++++";
  @Override
  public void fly() {
    Log.e(t, "animal fly");
  }

  @Override
  public void run() {
    Log.e(t, "animal run");
  }
}

File: MyReivces/app/src/main/java/com/fire/myreivces/ui/designMod/delegate/Animal.kt
package com.fire.myreivces.ui.designMod.delegate

import android.util.Log

interface Animal {

  fun back()
}

class Cat : Animal{
  override fun back() {
    Log.e("++++", "嘿嘿")
  }

}

class Zoo(animal: Animal): Animal by animal{

}